libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -504,6 +505,123 @@ TEST_F(CppInterface, PreprocessingOnlyModeSecondOverridesDisassemblyMode) {
               HasSubstr("void main(){ }"));
 }
 
+// To test file inclusion, use hashmap as a fake file system to store fake files
+// to be included.
+using FakeFS = std::unordered_map<std::string, std::string>;
+
+// A includer test case needs: 1) A fake file system which is actually an
+// unordered_map, so that we can resolve the content given a file path. 2) An
```

Comment:
Also, should describe the special role of the "root" file. That way you don't need to (and shouldn't) later in the test code.